src/cryptonote_basic/cryptonote_basic_impl.cpp

```diff
@@ -138,42 +138,26 @@ template bool get_dev_fund_amount<TESTNET>(uint64_t height, uint64_t& amount);
 template bool get_dev_fund_amount<STAGENET>(uint64_t height, uint64_t& amount);
 
 //-----------------------------------------------------------------------------------------------
-bool get_block_reward(network_type nettype, size_t median_size, size_t current_block_size, uint64_t already_generated_coins, uint64_t &reward, uint64_t height)
+uint64_t get_base_reward_tabular(uint64_t height, uint64_t already_generated_coins)
 {
-	uint64_t base_reward;
-	uint64_t round_factor = 10000000; // 1 * pow(10, 7)
-	if(height > 0)
-	{
-		if(height < (PEAK_COIN_EMISSION_HEIGHT + COIN_EMISSION_HEIGHT_INTERVAL))
-		{
-			uint64_t interval_num = height / COIN_EMISSION_HEIGHT_INTERVAL;
-			double money_supply_pct = 0.1888 + interval_num * (0.023 + interval_num * 0.0032);
-			base_reward = ((uint64_t)(MONEY_SUPPLY * money_supply_pct)) >> EMISSION_SPEED_FACTOR;
-		}
-		else
-		{
-			base_reward = (MONEY_SUPPLY - already_generated_coins) >> EMISSION_SPEED_FACTOR;
-		}
-	}
-	else
-	{
+	uint64_t BLOCK_REWARDS[] = {32000000000, 36450000000, 41970000000, 48590000000, 56280000000, 65070000000, 65070000000, 60820000000, 56840000000, 53130000000, 49660000000, 46420000000, 43390000000, 40550000000, 37910000000, 35430000000, 33120000000, 30950000000, 28930000000, 27040000000, 25280000000, 23630000000, 22080000000, 20640000000, 19290000000, 18030000000, 16850000000, 15750000000, 14720000000, 13760000000, 12860000000, 12020000000, 11240000000, 10500000000, 9820000000, 9180000000, 8580000000, 8020000000, 7490000000, 7000000000, 6540000000, 6120000000, 5720000000, 5340000000, 4990000000, 4670000000, 4360000000, 4080000000};
+
+	uint64_t interval_num = std::min(uint64_t(height / COIN_EMISSION_HEIGHT_INTERVAL), uint64_t(sizeof(BLOCK_REWARDS)/sizeof(uint64_t) - 1llu));
```

Comment:
This `min` function will return an int that is `0 <= interval_num <= 47` correct? It looks like it will but it is important enough to double check.

Reply:
Yes the result will be always be in the range of the array